crates/ruma-common/src/events/room/encrypted/relation_serde.rs

```diff
@@ -18,26 +18,22 @@ impl<'de> Deserialize<'de> for Relation {
         let ev = EventWithRelatesToJsonRepr::deserialize(deserializer)?;
 
         #[cfg(feature = "unstable-msc3440")]
-        if let Some(RelationJsonRepr::ThreadStable(ThreadStableJsonRepr {
-            event_id,
-            is_falling_back,
-        })) = ev.relates_to.relation
-        {
-            let in_reply_to = ev
-                .relates_to
-                .in_reply_to
-                .ok_or_else(|| serde::de::Error::missing_field("m.in_reply_to"))?;
-            return Ok(Relation::Thread(Thread { event_id, in_reply_to, is_falling_back }));
-        } else if let Some(RelationJsonRepr::ThreadUnstable(ThreadUnstableJsonRepr {
-            event_id,
-            is_falling_back,
-        })) = ev.relates_to.relation
-        {
-            let in_reply_to = ev
-                .relates_to
-                .in_reply_to
-                .ok_or_else(|| serde::de::Error::missing_field("m.in_reply_to"))?;
-            return Ok(Relation::Thread(Thread { event_id, in_reply_to, is_falling_back }));
+        match ev.relates_to.relation {
+            Some(RelationJsonRepr::ThreadStable(ThreadStableJsonRepr {
+                event_id,
+                is_falling_back,
+            }))
+            | Some(RelationJsonRepr::ThreadUnstable(ThreadUnstableJsonRepr {
```

Comment:
```suggestion
            })
            | RelationJsonRepr::ThreadUnstable(ThreadUnstableJsonRepr {
```

Reply:
Oh I didn't know it worked like that!

Reply:
It's a pretty recent thing, stable as of 1.53: https://caniuse.rs/features/or_patterns